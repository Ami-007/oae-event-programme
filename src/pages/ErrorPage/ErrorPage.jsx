import React from "react";
import "./ErrorPage.scss";
import sadNote from "../../assets/images/sad-note.svg";

const ErrorPage = () => {
  return (
    <div className="error-page">
      <h1 className="error-page__heading">
        Oops. Looks like this page doesnt exist. <br />
        <span className="error-page__heading--span">404</span>
      </h1>
      <img className="error-page__icon" src={sadNote} alt="sad musical note" />

      <a className="error-page__button" href="//oae.co.uk/">
        Back to OAE website
      </a>
    </div>
  );
};

export default ErrorPage;
