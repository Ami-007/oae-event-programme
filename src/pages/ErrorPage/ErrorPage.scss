.error-page {
  height: 100vh;
  background-color: rgb(103, 96, 212);
  text-align: center;
  overflow: scroll;
  padding: 2rem;

  &__heading {
    color: white;
  }

  &__subheading {
    color: white;
  }

  &__button {
    height: 66px;
    max-width: 20rem;
    text-transform: uppercase;
    background-color: #f97850;
    border-radius: 5px;
    color: white;
    font-weight: 700;

    font-size: 1rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-decoration: none;

    margin: 38px auto;
  }
}
