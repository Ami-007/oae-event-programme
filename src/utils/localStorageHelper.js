export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
