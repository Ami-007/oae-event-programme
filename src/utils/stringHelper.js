export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
