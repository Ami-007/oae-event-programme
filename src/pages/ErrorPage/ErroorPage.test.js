import { customRender } from "../../utils/testUtils";
import ErrorPage from "./ErrorPage";

it("should render the error page", () => {
  const { container } = customRender(<ErrorPage />);

  expect(container).toMatchSnapshot();
});
