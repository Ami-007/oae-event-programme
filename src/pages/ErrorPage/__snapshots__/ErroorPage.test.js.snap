// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the error page 1`] = `
<div>
  <div
    class="error-page"
  >
    <h1
      class="error-page__heading"
    >
      Oops. Looks like this page doesnt exist. 
      <br />
      <span
        class="error-page__heading--span"
      >
        404
      </span>
    </h1>
    <img
      alt="sad musical note"
      class="error-page__icon"
      src="sad-note.svg"
    />
    <a
      class="error-page__button"
      href="//oae.co.uk/"
    >
      Back to OAE website
    </a>
  </div>
</div>
`;
